DBSCE/Overview.md

```diff
@@ -0,0 +1,221 @@
+- MacOS: TBD
+- Android:TBD
+
+Note: Above are examples of Local Key Helpers that can be used for DBSC(E) key generation. Any platform is free to build a version of this, as long as the API is consistent with the protocol.
+
+### Attestation Service:
+
+A service that is responsible for verifying that the binding key is issued by the expected device and providing the attestation of the key to the IdP. The attestation service can be owned by the IdP or a third party. DBSC(E) relies on the attestation service to validate the binding statement and ensure that the binding key and the device key belong to the same device. We have added details on the specifics of the binding artifacts generated during the device registration process, and the validation of the binding statement in the [DBSC(E) Key Generation](#key-generation-specifics) section.
+
+### Device Registration (Pre-Session)
+
+Any enterprise user is expected to either use a device issued by their organization or register their personal device with the organization. The device registration is expected to be a once-in-a-lifetime operation, and the user is expected to perform this operation in a clean room environment.
+
+![DeviceRegistration](./DeviceRegistration.svg)
+
+## High-Level Design
+
+DBSC(E), if enabled for a given enteprise, specifies the generation of the cryptographic artifacts (keys and binding info) before a sign in session is established. By enabling the browser to invoke specific APIs based on an existing policy, it allows enterprises to add to the existing key generation. It also allows them to place stricter restrictions on specific sessions, hence providing the flexibility to secure a session appropriately.
+
+The high-level design is divided into two parts:
+
+1. Key generation and validation before the session starts (DBSC(E) is focused on this part).
+2. DBSC protocol applied with the generated keys (DBSC is focused on this part).
+
+Since we want to integrate DBSC(E) with the original design and make it as widely applicable as possible for all enterprise users, we are adding high-level design for the most possible combinations in this document. The intent is to have a specification that can be implemented by any browser vendor, and can be used by any IdP, and any Local Key Helper. As we cover different use cases DBSC(E) can be applied for, we differentiate between private and public local key helpers, since there are implications to the protocol based on the type of local key helper. For example, we expect well establised IdPs like Microsoft, Okta, Github to ship their own local key helper as a part of the IdP. We also optimize the DBSC(E) protocol for RP and IdP as the same service (google client authenticated by google service as an example), since it simplifies the protocol and aligns with the perf goals for specific services.
+
+DBSC(E) (in contrast with DBSC):
+
+![DBSC(E) Highlevel Design](<./DBSC(E).svg>)
```

Comment:
Some minor diagram feedback:

1. It's confusing on this diagram that initiate sign-in and response with Sec-Session-GenerateKey and HelperIdList are 2 separate requests, and the browser is the one that initiates sign-in. Should we combine those two like we did in draft diagrams?

2. Comment after step nr 2 that "browser pre-generates keys based on the headers" isn't precise. Browser doesn't pre-generate those keys. It calls LocalKeyHelper to pre-generate those keys. 

3. Step 3 - serverURL is one parameter, but we need to be split into 2 (RP, IDP)

4. LocalKeyHelper/TPM actor should be just LocalKeyHelper

5. Step 7 needs to return KeyId

6. Step 10 will not always be 200 response, should we remove it to avoid confusion and maybe replace with something like "Successful response"?

7. Step 11 should have KeyId

Reply:
Let us discuss this offline before I make changes. I kept some of the DBSC design to draw contrast on what changed in DBSC(E). For example, DBSC shows browser initiating the sign in, and mentions TPM etc.

Reply:
Made changes.